src/core/Bot.py

```diff
@@ -261,19 +264,32 @@ async def db_latency(self):
         return f"{t2*1000:.2f} ms"
 
     @staticmethod
-    async def getch(get_method, fetch_method, _id):  # why does c have all the fun?
+    async def getch(get_method, fetch_method, _id) -> Any:  # why does c have all the fun?
         try:
             _result = get_method(_id) or await fetch_method(_id)
         except (discord.HTTPException, discord.NotFound):
             return None
         else:
             return _result
 
+    async def get_or_fetch_message(self, channel: discord.Channel, message_id: int) -> Optional[discord.Message]:
+        # caching cause, due to rate limiting 50/1
+        try:
+            return self.message_cache[message_id]
+            # scripting is always fater than `.get()`
```

Comment:
Another reason is, caching won't ever work if using `.get()`. As it always return None. And hence it force us to recheck. And other `if-else` clause